document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AccountsHandlerImpl.java

```diff
@@ -1,13 +1,56 @@
 package uk.gov.companieshouse.document.generator.accounts.handler.accounts;
 
+import static uk.gov.companieshouse.document.generator.accounts.AccountsDocumentInfoServiceImpl.MODULE_NAME_SPACE;
+
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.model.accounts.Accounts;
+import uk.gov.companieshouse.document.generator.accounts.AccountType;
+import uk.gov.companieshouse.document.generator.accounts.LinkType;
+import uk.gov.companieshouse.document.generator.accounts.exception.HandlerException;
+import uk.gov.companieshouse.document.generator.accounts.exception.ServiceException;
+import uk.gov.companieshouse.document.generator.accounts.service.AccountsService;
 import uk.gov.companieshouse.document.generator.interfaces.model.DocumentInfoResponse;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
 
 @Component
 public class AccountsHandlerImpl implements AccountsHandler  {
 
+    private static final Logger LOG = LoggerFactory.getLogger(MODULE_NAME_SPACE);
+
+    @Autowired
+    AccountsService accountsService;
+
     @Override
-    public DocumentInfoResponse getAccountsData(String resourceLink) {
+    public DocumentInfoResponse getAccountsData(String resourceLink) throws HandlerException {
+        Accounts accounts;
+
+        try {
+            accounts = accountsService.getAccounts(resourceLink);
+        } catch (ServiceException e) {
+            LOG.error(e);
```

Comment:
No need to log here; this exception is already logged at the controller

Reply:
removed